core/src/test/java/io/grpc/internal/DnsNameResolverTest.java

```diff
@@ -444,8 +468,9 @@ public void resolve_cacheExpired() throws Exception {
     String name = "foo.googleapis.com";
     FakeTicker fakeTicker = new FakeTicker();
 
-    DnsNameResolver resolver =
-        newResolver(name, 81, GrpcUtil.NOOP_PROXY_DETECTOR, Stopwatch.createUnstarted(fakeTicker));
+    DnsNameResolver resolver = (DnsNameResolver) newResolver(
+        name, 81, GrpcUtil.NOOP_PROXY_DETECTOR,
+        Stopwatch.createUnstarted(fakeTicker)).getRetriedNameResolver();
```

Comment:
Why aren't you using the RetryingNameResolver for the shutdown?  Seems like getting the delegate makes mistakes much easier.

Reply:
Good point, that would be way more precise. I updated all the tests to only use DnsNameResolver when they specifically need its API.